### PR TITLE
No base64 encoding; Retry on syncing Conversations

### PR DESCRIPTION
**Pull request by michaelkantor**
1. The Encoding property has been removed
    * Now insures that textual Parts are delivered as null or string
    * Now insures that non-textual parts are delivered as Blob
    * No special case for base64 encoded data
2. Retry on syncing Conversations [WEB-1053]
    * layer.Query will retry every 1.5-2 seconds until a Conversation is either synced or returning the desired number of results.

### CODE REVIEW

File: src/query.js [hunk 1]
```diff
@@ -588,6 +588,9 @@ class Query extends Root {
     if (results.success) {
       this._appendResults(results, false);
       this.totalSize = Number(results.xhr.getResponseHeader('Layer-Count'));
+      if (results.xhr.getResponseHeader('Layer-Conversation-Is-Syncing') === 'true' && this.data.length < this.paginationWindow) {
```
Comment (michaelkantor) [inline src/query.js]: Compare to master; needs a server-syncing-state event.

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,13 @@
 * No more `temp_layer:///` IDs, no more ID change events
   * Sending a Message will no longer result in a `messages:change` event that contains a change to the Message ID; the Message ID that is assigned when creating the Message will be accepted by the server.
   * Conversations may still get an ID change event in the case where a Distinct Conversation is created, and a matching Distinct Conversation is found on the server.
+* Simplified access to MessagePart data; `MessagePart.body` will be either a string or Blob, never a base64 string
```
Comment (nilgradisnik) [inline CHANGELOG.md]: This should be moved to Beta 2 changelog.

File: src/message-part.js [hunk 3]
```diff
@@ -105,9 +105,26 @@ class MessagePart extends Root {
...
+
+    // Custom handling for non-textual content
+    if (!textual) {
+      // If the body exists and is a blob, extract the data uri for convenience; only really relevant for image and video HTML tags.
+      if (isBlobBody) {
+        this.url = URL.createObjectURL(this.body);
+      }
+
+      // If we have a body (not rich content), and our non-textual content is not a blob, replace the body with a blob
+      else if (this.body && !isBlobBody && !textual) {
```
Comment (nilgradisnik) [inline src/message-part.js]: `!isBlobBody && !textual` is unnecessary

File: src/query.js [hunk 4]
```diff
@@ -369,8 +371,9 @@ class Query extends Root {
       const removedData = this.data.slice(this.paginationWindow);
       this.data = this.data.slice(0, this.paginationWindow);
       this.client._checkAndPurgeCache(removedData);
+      this.pagedToEnd = false;
       this._triggerAsync('change', { data: [] });
-    } else if (pageSize === 0) {
+    } else if (pageSize === 0 || this.pagedToEnd) {
```
Comment (michaelkantor) [inline src/query.js]: Alternate property names?

**Pull request merged**